src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+}
+
+static void SetMickeyPixelRate(const int16_t ratio_x, const int16_t ratio_y)
+{
+    // According to https://www.stanislavs.org/helppc/int_33-f.html
+    // the values should be non-negative (highest bit not set)
+
+    if ((ratio_x > 0) && (ratio_y > 0)) {
+        constexpr auto X_MICKEY = 8.0f;
+        constexpr auto Y_MICKEY = 8.0f;
+
+        state.mickeys_per_px_x = static_cast<float>(ratio_x) / X_MICKEY;
+        state.mickeys_per_px_y = static_cast<float>(ratio_y) / Y_MICKEY;
+        state.pxs_per_mickey_x = X_MICKEY / static_cast<float>(ratio_x);
+        state.pxs_per_mickey_y = Y_MICKEY / static_cast<float>(ratio_y);
+    }
+}
+
+static void SetDoubleSpeedThreshold(uint16_t threshold)
+{
+    if (threshold)
+        state.dspeed_threshold = threshold;
+    else
+        state.dspeed_threshold = 64; // default value
+}
+
+static void SetSensitivity(uint16_t px, uint16_t py, uint16_t dspeed_threshold)
+{
+    px               = std::min(static_cast<uint16_t>(100), px);
+    py               = std::min(static_cast<uint16_t>(100), py);
+    dspeed_threshold = std::min(static_cast<uint16_t>(100), dspeed_threshold);
+    // save values
+    state.senv_x_val       = px;
+    state.senv_y_val       = py;
+    state.dspeed_threshold = dspeed_threshold;
+    if ((px != 0) && (py != 0)) {
+        px--; // Inspired by CuteMouse
+        py--; // Although their cursor update routine is far more
+              // complex then ours
+        state.senv_x = (static_cast<float>(px) * px) / 3600.0f + 1.0f / 3.0f;
+        state.senv_y = (static_cast<float>(py) * py) / 3600.0f + 1.0f / 3.0f;
+    }
+}
+
+static void SetInterruptRate(uint16_t rate_id)
+{
+    switch (rate_id) {
+    case 0: rate_hz  = 0;   break; // no events, TODO: this should be simulated
+    case 1: rate_hz  = 30;  break;
+    case 2: rate_hz  = 50;  break;
+    case 3: rate_hz  = 100; break;
+    default: rate_hz = 200; break; // above 4 is not suported, set max
+    }
+
+    // Update event queue settings
+    if (rate_hz) {
+        // Update event queue settings
+        MOUSE_NotifyRateDOS(rate_hz);
+    }
+}
+
+static void ResetHardware()
+{
+    wheel = 0;
+    SetInterruptRate(4);
+    PIC_SetIRQMask(12, false); // lower IRQ line
+}
+
+void MOUSEDOS_BeforeNewVideoMode()
+{
+    if (CurMode->type != M_TEXT)
+        RestoreCursorBackground();
+    else
+        RestoreCursorBackgroundText();
+
+    state.hidden     = 1;
+    state.oldhidden  = 1;
+    state.background = false;
+}
+
+// TODO: Does way to much. Many things should be moved to mouse reset one day
+void MOUSEDOS_AfterNewVideoMode(const bool setmode)
+{
+    state.inhibit_draw = false;
+    // Get the correct resolution from the current video mode
+    uint8_t mode = mem_readb(BIOS_VIDEO_MODE);
+    if (setmode && mode == state.mode)
+        LOG(LOG_MOUSE, LOG_NORMAL)
+    ("New video mode is the same as the old");
+    state.gran_x = 0xffff;
+    state.gran_y = 0xffff;
```

Comment:
what is 0xffff?

Reply:
No idea - legacy code without comments, I haven't analyzed it yet.